waku/protocol/v2/waku_store.nim

```diff
@@ -78,11 +80,18 @@ proc encode*(rpc: HistoryRPC): ProtoBuffer =
 
 proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse =
   result = HistoryResponse(messages: newSeq[WakuMessage]())
-  for msg in w.messages:
-    if msg.contentTopic in query.topics:
-      result.messages.insert(msg)
+
+  let topics = join(query.topics, ", ")
+  for row in w.db.rows(SqlQuery("SELECT contentTopic, payload FROM messages WHERE contentTopic IN (" & topics & ")")):
+    result.messages.add(WakuMessage(payload: cast[seq[byte]](row[1]), contentTopic: ContentTopic(row[0].parseInt)))
 
 method init*(ws: WakuStore) =
+  ws.db.exec(sql"""CREATE TABLE IF NOT EXISTS messages (
```

Comment:
I'd add a comment explaining semantics briefly. Aslo might want to break out sql statement into constant.